packages/bsky/src/hydration/hydrator.ts

```diff
@@ -383,13 +398,19 @@ export class Hydrator {
     const allPostUris = [...posts.keys()]
     const allRefs = [
       ...refs,
-      ...postUrisLayer1.map(uriToRef), // supports aggregates on embed #viewRecords
-      ...postUrisLayer2.map(uriToRef),
+      ...embedPostUrisLayer1.map(uriToRef), // supports aggregates on embed #viewRecords
+      ...embedPostUrisLayer2.map(uriToRef),
     ]
     const threadRefs = allRefs.map((ref) => ({
       ...ref,
       threadRoot: posts.get(ref.uri)?.record.reply?.root.uri ?? ref.uri,
     }))
+    const postUrisWithPostgates = new Set<string>()
+    for (const [uri, post] of posts) {
+      if (post && post.hasPostGate) {
+        postUrisWithPostgates.add(uri)
+      }
+    }
```

Comment:
If this is too complicated feel free to ignore, but I _think_ we only need postgates on nested posts, not necessarily for e.g. `postsLayer0`.  That's going to be a lot fewer to check since most posts don't have a post embed.  I believe that would be `[...postUrisLayer1, ...postUrisLayer2]`.

Reply:
Postgates also power `ViewerState.embeddingDisabled`, so I think we do unfortunately need it for all posts, even top level :/

Reply:
Makes sense!